public/app/plugins/datasource/grafana-azure-monitor-datasource/credentials.ts

```diff
@@ -116,7 +115,8 @@ export function getCredentials(options: AzureDataSourceSettings): AzureCredentia
   }
 }
 
-export function getLogAnalyticsCredentials(options: AzureDataSourceSettings): AzureCredentials | undefined {
+/** @deprecated since 8.0, left here for help the migration */
+export function getLogAnalyticsCredentials(options: AzureDataSourceSettings): AzureClientSecretCredentials | undefined {
```

Comment:
I'm not sure if I follow the reason for the changes to this function and the `AzureClientSecretCredentials` type.

I think that it would be fine to keep it with the old behaviour (or change it to only make it return credentials if sameAs is false) and then anywhere other than the read-only credentials view should update to use `getCredentials`.

Then we can just mark the function as deprecated (with a slightly more descriptive message indicating separate credentials are no longer used, and its just for info purposes in config editor only), and leave the `AzureClientSecretCredentials` type alone.